promoter/image/promoter.go

```diff
@@ -120,15 +125,31 @@ func (p *Promoter) PromoteImages(opts *options.Options) (err error) {
 		return nil
 	}
 
+	// Verify any signatures in staged images
+	if err := p.impl.ValidateStagingSignatures(
+		opts, sc, promotionEdges,
+	); err != nil {
+		return errors.Wrap(err, "checking signtaures in staging images")
+	}
+
 	// Check the pull request
 	if !opts.Confirm {
 		return p.impl.PrecheckAndExit(opts, mfests)
 	}
 
-	return errors.Wrap(
-		p.impl.PromoteImages(sc, promotionEdges, producerFunc),
-		"running promotion",
-	)
+	if err := p.impl.PromoteImages(sc, promotionEdges, producerFunc); err != nil {
+		return errors.Wrap(err, "running promotion")
+	}
+
+	if err := p.impl.SignImages(opts, sc, promotionEdges); err != nil {
+		return errors.Wrap(err, "signing images")
+	}
+
+	if err := p.impl.WriteSBOMs(opts, sc, promotionEdges); err != nil {
```

Comment:
should we write the bom before signing the image?

Reply:
or does not matter?

Reply:
Right now there is no difference, but I've been thinking that the BOM could verify the image before generation, becoming the first consumer of the signature.